integration-tests/rest-client-reactive/src/main/java/io/quarkus/it/rest/client/main/ParamConverter.java

```diff
@@ -0,0 +1,47 @@
+package io.quarkus.it.rest.client.main;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.ext.ParamConverterProvider;
+
+public class ParamConverter implements ParamConverterProvider {
+    @SuppressWarnings("unchecked")
+    @Override
+    public <T> javax.ws.rs.ext.ParamConverter<T> getConverter(Class<T> rawType, Type genericType,
+            Annotation[] annotations) {
+        if (genericType == null || !genericType.equals(Param.class)) {
+            throw new RuntimeException("Generic Type cannot be null!");
```

Comment:
the messages do not conform to what is checked now but this can be fixed later, by the way

Reply:
I've changed the message to simply say `Wrong generic type..."